inputs/range.vue

```diff
@@ -0,0 +1,411 @@
+   * used for keyboard support
+   * @param  {number|array} val
+   * @param  {number} index
+   * @return {number}
+   */
+  function getNumber(val, index) {
+    if (_.isNumber(val)) {
+      return val;
+    } else if (_.isArray(val)) {
+      return val[index];
+    } else {
+      throw new Error('Unable to get value from range input!');
+    }
+  }
+
+  /**
+   * set range value back into input,
+   * used for keyboard support
+   * @param {number} val
+   * @param {number|array} rawVal
+   * @param {number} index
+   * @param {function} setter
+   */
+  function setNumber(val, rawVal, index, setter) {
+    if (_.isNumber(rawVal)) {
+      setter(val);
+    } else if (_.isArray(rawVal) && index === 0) {
+      setter([val, rawVal[1]]);
+    } else if (_.isArray(rawVal) && index === 1) {
+      setter([rawVal[0], val]);
+    } else {
+      throw new Error('Unable to set value into range input!');
+    }
+  }
+
+  export default {
+    props: ['name', 'data', 'schema', 'args'],
+    data() {
+      return {
+        values: this.data || this.isDualPoint ? [0, 0] : 0
+      };
+    },
+    computed: {
+      label() {
+        return label(this.name, this.schema);
+      },
+      start() {
+        return this.args.start || 0;
+      },
+      isDualPoint() {
+        return _.isArray(this.start);
+      },
+      step() {
+        return this.args.step || 1;
+      },
+      min() {
+        return this.args.min || 0;
+      },
+      max() {
+        return this.args.max || 10;
+      },
+      minLabel() {
+        return this.args.minLabel || this.min.toString();
+      },
+      maxLabel() {
+        return this.args.maxLabel || this.max.toString();
+      },
+      tooltips() {
+        return !!this.args.tooltips;
+      },
+      errorMessage() {
+        return getValidationError(this.data, this.args.validate, this.$store, this.name);
+      },
+      showError() {
+        return !!this.errorMessage;
+      },
+      hasFeedback() {
+        return this.args.help || this.showError;
+      },
+      rangeClasses() {
+        return {
+          'has-tooltips': this.args.tooltips
+        };
+      }
+    },
+    methods: {
+      update(values) {
+        // if we're dealing with a single value, grab it directly
+        // otherwise grab the whole array
+        const val = this.isDualPoint ? values : _.head(values);
+
+        this.$store.commit(UPDATE_FORMDATA, { path: this.name, data: val });
+      }
+    },
+    mounted() {
+      const el = find(this.$el, '.editor-range-input'),
+        step = this.step,
+        min = this.min,
+        max = this.max,
+        minLabel = this.minLabel,
+        maxLabel = this.maxLabel;
+
+      slider.create(el, {
+        start: this.start,
+        step: step,
+        range: { min, max },
+        format: {
+          to(val) {
+            return step >= 1 ? parseInt(val) : val;
+          },
+          from(val) {
+            return step >= 1 ? parseInt(val) : val;
```

Comment:
Can you get away with a simple typecast here? `Number(val)`

Reply:
fixed!